be/src/exprs/map_functions.cpp

```diff
@@ -124,6 +124,76 @@ StatusOr<ColumnPtr> MapFunctions::map_from_arrays(FunctionContext* context, cons
     }
 }
 
+
+StatusOr<ColumnPtr> MapFunctions::map_from_entries(FunctionContext* context, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto keys_column = ColumnHelper::unpack_and_duplicate_const_column(columns[0]->size(), columns[0]);
+    NullColumn* keys_null = nullptr;
+    ArrayColumn* keys_data = nullptr;
+    if (keys_column->is_nullable()) {
+        auto keys = down_cast<NullableColumn*>(keys_column.get());
+        keys_null = keys->null_column().get();
+        keys_data = down_cast<ArrayColumn*>(ColumnHelper::get_data_column(keys));
+    } else {
+        keys_data = down_cast<ArrayColumn*>(keys_column.get());
+    }
+
+    auto& keys_offsets = keys_data->offsets().get_data();
+
+    ColumnPtr& elements = keys_data->elements_column();
+    StructColumn* struct_column = nullptr;
+    if (elements->is_nullable()) {
+        auto keys = down_cast<NullableColumn*>(elements.get());
+        struct_column = down_cast<StructColumn*>(ColumnHelper::get_data_column(keys));
+    } else {
+        struct_column = down_cast<StructColumn*>(elements.get());
+    }
```

Comment:
```
    if (elements->is_nullable()) {
        auto keys = down_cast<NullableColumn*>(elements.get());
        struct_column = down_cast<StructColumn*>(ColumnHelper::get_data_column(keys));
    } else {
        struct_column = down_cast<StructColumn*>(elements.get());
    }
```

get_data_column can process nullable column and not nullable column internally, so just writing like this is ok 

```
auto*  struct_column = down_cast<StructColumn*>(elements.get());
```